en/developer_guide/bbcode.rst

```diff
@@ -0,0 +1,173 @@
+BBCode extension
+================
+
+In the Back Office, a Wysiwyg editor is used on several places where a rich text can be contributed.
+You can encounter it for instance on certain blocks or Content Types. Although most of the
+information entered in the editor is stored in the database in html format, you can also extend the
+possibilities by introducing custom BBCode tags. This is partially done via the Open Orchestra
+BBCode parser.
+
+It's important to note that by default, not all tags are enabled in BBCode format, only those you
+want to declare. For instance, putting a text in bold in the rich text editor does not produce by
+default a ``[b]`` tag. Out of the box, there is only one BBCode tag: ``media``. But it is possible
+to add your own tags.
+
+In this documentation you will see how the BBCode feature fits in the main processes and how to create
+a new BBCode in your bundle.
+
+Wysiwyg process
+---------------
+Open Orchestra uses TinyMce to render a rich text input. Normally, TinyMce generates html tags, but
+Open Orchestra has enhanced it to use BBCode. Here is the workflow triggered when editing an object
+using a form containing rich text fields:
+
+.. image:: ../../images/richtext2.png
+
+1. Each rich text field goes into a data transformer to interpret the BBCode tags. This data
+   transformer calls the Open Orchestra BBCode parser to detect and transform valid BBCode tags into
+   html tags.
+2. The resulting html is given to the TinyMce editor which treats it with its standard logic.
+3. When the user clicks in the toolbar to inject a BBCode tag (for instance by selecting a media),
+   TinyMce inject the html representation of the BBCode, to be able to visually render the tag. For
+   instance in the media case, you will see it in the editor.
+4. When the user saves the form, TinyMce replaces via javascript the html tags by their BBCode
+   representations.
+
+How to create a new BBCode and add it to the parser
+---------------------------------------------------
+Open Orchestra provides a simple ways of adding your own custom BBCode tags, depending on their
+nature. Two kind of tags exists:
+
+- Simple tags are just text replacements. For instance a ``[b]Some Text[/b]`` should be replaced by
+``<b>Some Text</b>``
+
+- Complex tags require more than a simple regexp conversion. For instance the tag
+``[media=format]MediaId[/media]`` requires to fetch the media in the database, to detect its type
+and then to to return the corresponding html tag.
+
+To add a new BBCode to the parser, you need to create its definition. There are two ways of adding
+a definition. To create a simple tag you can quickly set a configuration. But if you prefer, or if
+your tag requires a php analysis, you can use a tagged service.
```

Comment:
you can use a symfony service (I don't think that you have to already say that the service should be tagged, it will be said later)